scc/cert/bitset.go

```diff
@@ -62,3 +64,20 @@ func (b BitSet[T]) String() string {
 type unsigned interface {
 	~uint8 | ~uint16 | ~uint32 | ~uint64
 }
+
+// MarshalJSON converts the BitSet into a JSON-compatible hex string.
+func (b *BitSet[T]) MarshalJSON() ([]byte, error) {
+	return json.HexBytes(b.mask).MarshalJSON()
+}
+
+// UnmarshalJSON parses a JSON hex string into a BitSet.
+func (b *BitSet[T]) UnmarshalJSON(data []byte) error {
+	hexBytes := json.HexBytes{}
+	err := hexBytes.UnmarshalJSON(data)
+	if err != nil {
+		return err
+	}
+	b.mask = make([]byte, len(hexBytes))
+	copy(b.mask, hexBytes)
```

Comment:
nit: no need to copy
```suggestion
	b.mask = hexBytes[:]
```